app/services/users/omniauth.rb

```diff
@@ -1,33 +1,26 @@
-class Users::Omniauth
-  attr_reader :auth
-
-  def initialize(auth)
-    @auth = auth
-  end
-
+class Users::Omniauth < Struct.new(:auth)
   def get
-    user
+    authorization_create
   end
 
   private
 
   delegate :provider, :uid, :info, to: :auth
   delegate :email, to: :info
 
-  def user
-    user = User.where(provider: provider, uid: uid, email: email).first
-    user ||= User.new(
-      provider: provider,
-      uid: uid,
-      email: email,
-      password: Devise.friendly_token[0, 20]
-    )
-    user_create(user)
+  def authorization_create
+    authorization = user.authorizations.create_with(uid: uid).find_or_create_by(provider: provider)
+    authorization.user
   end
```

Comment:
@marderer `#create_authorization` sounds like it creates authorization, but from your code, it returns `user` - it's unexpectable